File: works/2705.py
d = [0 for _ in range(1010)]
d[1] = 1
d[2] = 2

for i in range(1, 1001):
    sum = 0
    for j in range(1, i//2+1):
        sum += d[j]
    d[i] = sum + 1

t = int(input())
while(t > 0):
    t -= 1
    n = int(input())
    print(d[n])

File: works/2705.cs
using System;
using System.Collections;
using static System.Console;

namespace ConsoleApp
{
    class Program
    {
        static void Main(string[] args)
        {
            int[] d = new int[1010];
            d[1] = 1;
            d[2] = 2;
            for (int i = 1; i <= 1000; i++)
            {
                int sum = 0;
                for (int j = 1; j <= i / 2; j++) sum += d[j];
                d[i] = sum + 1;
            }
            int t = int.Parse(ReadLine());
            while (t-- > 0)
            {
                int n = int.Parse(ReadLine());
                WriteLine(d[n]);
            }
        }
    }

}
